presto-record-decoder/src/main/java/io/prestosql/decoder/raw/RawColumnDecoder.java

```diff
@@ -202,31 +201,32 @@ public FieldValueProvider decodeField(byte[] value)
                     columnName,
                     value.length));
         }
-        return new RawValueProvider(ByteBuffer.wrap(value, start, actualEnd - start), fieldType, columnName, columnType);
+        return new RawValueProvider(Slices.wrappedBuffer(value, start, actualEnd - start), fieldType, columnName, columnType);
     }
 
     private static class RawValueProvider
             extends FieldValueProvider
     {
-        private final ByteBuffer value;
+        private final Slice value;
         private final FieldType fieldType;
         private final String columnName;
         private final Type columnType;
-        private final int size;
+        private final int length;
+        private final int ZERO_OFFSET = 0;
```

Comment:
inline

Reply:
do you mean replace `getLong(ZERO_OFFSET)` with `getLong(0)`?

Reply:
yes